modular_skyrat/modules/gun_cargo/code/objects/firing_pin.dm

```diff
@@ -0,0 +1,39 @@
+GLOBAL_VAR_INIT(permit_pin_unrestricted, FALSE)
+// Firing pin that can be used off station freely, and requires a permit to use on-station
+/obj/item/firing_pin/permit_pin
+	name = "permit-locked firing pin"
+	desc = "A firing pin for a station who can't trust their crew. Only allows you to fire the weapon off-station or with a firearms permit.."
+	icon_state = "firing_pin_explorer"
+	fail_message = "<span class='warning'>You must have a permit or be off-station to fire this!</span>"
+	can_remove = FALSE
+
+// This checks that the user isn't on the station Z-level.
+/obj/item/firing_pin/permit_pin/pin_auth(mob/living/user)
+	var/turf/station_check = get_turf(user)
+
+	if(obj_flags & EMAGGED)
+		return TRUE
+
+	if(GLOB.permit_pin_unrestricted)
+		return TRUE
+
+	if(!ishuman(user))
+		return FALSE
+
+	var/mob/living/carbon/human/user_human = user
+
+	var/obj/item/card/id/the_id = user_human.wear_id
```

Comment:
We have a fairly robust get_idcard() proc for mobs of all types. You should be using even if just that for consistency.
```suggestion
	var/obj/item/card/id/the_id = user.get_idcard()
```

Reply:
do this

Reply:
It seems you have forgotten about this suggested change. Consider this comment a reminder of it.

Reply:
Resolved